exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -267,3 +294,58 @@ func timestampToSecondsWithMillisecondPrecision(ts pcommon.Timestamp) *float64 {
 func float64ToDimValue(f float64) string {
 	return strconv.FormatFloat(f, 'g', -1, 64)
 }
+
+// merge metric events to adhere to the multimetric format event.
+func mergeEventsToMultiMetricFormat(events []*splunk.Event) ([]*splunk.Event, error) {
+	merges := map[uint32][]int{}
+	hasher := fnv.New32a()
+	for i, e := range events {
+		cloned := copyEventWithoutValues(e)
+		data, err := jsoniter.Marshal(cloned)
+		if err != nil {
+			return nil, err
+		}
+		_, err = hasher.Write(data)
+		if err != nil {
+			return nil, err
+		}
+		hashed := hasher.Sum32()
+		hasher.Reset()
+		matches := merges[hashed]
+		if len(matches) == 0 {
+			merges[hashed] = []int{i}
+		} else {
+			merges[hashed] = append(merges[hashed], i)
+		}
+	}
+	var mergedEvents []*splunk.Event
+
+	for _, merge := range merges {
+		newEvent := events[merge[0]]
+		for j := 1; j < len(merge); j++ {
+			// check no collision for this event values:
+			eventValues := map[string]interface{}{}
+			collisionFound := false
```

Comment:
Are these two variables really needed? I might be missing something, but they seem redundant to me

Reply:
I clarified some of the code as the variable names feel a bit arcane, let me know if that helps clarify